lib/rules/strict.js

```diff
@@ -82,46 +82,46 @@ module.exports = function(context) {
     }
 
     //--------------------------------------------------------------------------
-    // "never" mode
+    // If this is modules, all "use strict" directives are unnecessary.
     //--------------------------------------------------------------------------
 
-    if (mode === "never") {
+    if (context.parserOptions.sourceType === "module") {
```

Comment:
This rule is selecting returning object by the option, so I think it introduces big complexity to this rule if it uses `sourceType` of `Program` node.